ksql-rest-app/src/test/java/io/confluent/ksql/rest/client/KsqlRestClientFunctionalTest.java

```diff
@@ -68,22 +75,40 @@
 public class KsqlRestClientFunctionalTest {
 
   private MockApplication mockApplication;
+  private ExecutableServer<KsqlRestConfig> mockServer;
   private KsqlRestClient ksqlRestClient;
 
   @Rule
   public final ExpectedException expectedException = ExpectedException.none();
 
   @Before
   public void init() throws Exception {
-    mockApplication = new MockApplication();
-    mockApplication.start();
+    final Map<String, Object> props = ImmutableMap.<String, Object>builder()
+        .put(KsqlRestConfig.LISTENERS_CONFIG, "http://localhost:0")
+        .put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092")
+        .put(RestConfig.SHUTDOWN_GRACEFUL_MS_CONFIG, (int) TimeUnit.SECONDS.toMillis(30))
+        .build();
+
+    final KsqlRestConfig ksqlRestConfig = new KsqlRestConfig(props);
+    mockApplication = new MockApplication(ksqlRestConfig);
+    mockServer = new ExecutableServer<>(
+        new ApplicationServer<>(ksqlRestConfig),
+        ImmutableList.of(mockApplication)
+    );
+
+    mockServer.startAsync();
 
     ksqlRestClient = buildClient(mockApplication.getServerAddress());
   }
 
   @After
   public void cleanUp() {
-    mockApplication.stop();
+    try {
```

Comment:
Why was this try catch added here?

Reply:
it used to be inside the `stop` method, but was removed when I implemented `triggerShutdown` instead, so it got pushed up one layer